src/environmentd/src/lib.rs

```diff
@@ -286,6 +292,38 @@ pub async fn serve(config: Config) -> Result<Server, anyhow::Error> {
     // Initialize controller.
     let controller = mz_controller::Controller::new(config.controller, envd_epoch).await;
 
+    // Initialize the system parameter frontend if `launchdarkly_sdk_key` is set.
+    let system_parameter_frontend = {
+        let ld_sdk_key = config.launchdarkly_sdk_key;
+        let ld_user_key = config
+            .frontegg
+            .as_ref()
+            .map(|frontegg| frontegg.tenant_id().to_string())
+            .unwrap_or_else(|| "anonymous-dev@materialize.com".to_string());
+
+        // The `SystemParameterFrontend::new` call needs to be wrapped in a
+        // spawn_blocking because the LaunchDarkly SDK initialization uses
+        // `reqwest::blocking::client`. This should be revisited after the
+        // SDK is updated to 1.0.0.
+        task::spawn_blocking(
+            || "SystemParameterFrontend::new",
+            move || {
+                ld_sdk_key
+                    .map(|ld_sdk_key| {
+                        SystemParameterFrontend::new(ld_sdk_key.as_str(), ld_user_key.as_str())
+                    })
+                    .transpose()
+                    .unwrap_or_else(|e| {
+                        tracing::error!("Cannot initialize system parameter frontend: {:?}", e);
```

Comment:
This is surprising to me. AFAICT `SystemParameterFrontend::new` just builds some structs, and any errors are usage errors that we can just return with `?`, rather than via a `tracing::error!`. What am I missing?

Reply:
Yes, this is a stupid leftover from a previous iteration when the constructor was also initializing the LD client.